src/config/config.ts

```diff
@@ -0,0 +1,351 @@
+import convict from "convict"
+
+convict.addFormat({
+  name: "required-string",
+  validate: (val: any) => {
+    if (!val) throw new Error("value cannot be empty, null or undefined")
+    if (typeof val !== "string") throw new Error("value must be a string")
+  },
+})
+
+convict.addFormat({
+  name: "required-positive-number",
+  validate: (val: any) => {
+    if (!val) throw new Error("value cannot be empty, null or undefined")
+    if (typeof val !== "number") throw new Error("value must be a number")
+  },
+  coerce: (val: string) => {
+    const coercedVal = Number(val)
+    if (coercedVal < 0) {
+      throw new Error("value must be a positive number")
+    }
+    return coercedVal
+  },
+})
+
+convict.addFormat({
+  name: "required-boolean",
+  validate: (val: any) => {
+    if (val === null || val === undefined)
+      throw new Error("value cannot be empty, null or undefined")
+    if (typeof val !== "boolean") throw new Error("value must be a boolean")
+  },
+  coerce: (val: string) => String(val).toLowerCase() === "true",
+})
+
+// Define a schema
+const config = convict({
+  env: {
+    doc: "The application environment.",
+    env: "NODE_ENV",
+    format: ["LOCAL_DEV", "DEV", "test", "prod"],
```

Comment:
need to add staging also, no? we might also use this chance to eliminate `local_dev` (since we don't have a diff eb setup separately for dev)

Reply:
our staging is using "DEV" as the value. But I will add in another option

Reply:
dun add, we shud eliminate the `local_dev` option cos it's useless HAHA

Reply:
@seaerchin i won't suggest to remove it in this PR though. We do have some logic where its being used, e.g. in `src/services/identity/index.ts` and so on

Should do this clean up separately

Reply:
sure, that's fine too; can track in issue?

Reply:
Sure, it's here: https://github.com/isomerpages/isomercms-backend/issues/656